backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmTemplateFromConfigurationCommand.java

```diff
@@ -641,6 +641,18 @@ public AuditLogType getAuditLogTypeValue() {
 
     @Override
     protected void updateDiskSizeByQcowImageInfo(DiskImage diskImage, Guid storageId) {
+        Set<Guid> storageDomains = getParameters().getImageToAvailableStorageDomains().get(diskImage.getImageId());
+
+        if (storageDomains != null && !storageDomains.isEmpty()) {
+            // Try to use the target SD, otherwise fallback to one of the available SDs
+            // for the image
+            if (storageDomains.contains(getStorageDomainId())) {
+                storageId = getStorageDomainId();
```

Comment:
I personally do not like when the method assigns a new value to its argument (`storageId`).

Also, in the `ImportVmTemplateCommand` the method is called twice based on the `getParameters().isImagesExistOnTargetStorageDomain()` condition:
```
if (!getParameters().isImagesExistOnTargetStorageDomain()) {
    updateDiskSizeByQcowImageInfo(image);
} else {
    updateDiskSizeByQcowImageInfo(image, image.getStorageIds().get(0));
}
```
Is it intentional to overwrite the `storageId` for both cases?

Reply:
yes, the logic that is moved here is less relevant to this case (because the image is supposed to be on the source storage domain) but it's not harmful - we want to update the size of the image, it doesn't really matter where we take it from

Reply:
I think it would be more readable if the `initImportClonedTemplateDisks` just called `updateDiskSizeByQcowImageInfo(image)` and each class would handle that differently instead of passing an argument that gets overwritten, but if that works and fixes the issue, it can be merged.